src/eth_provider/starknet/kakarot_core.rs

```diff
@@ -142,14 +142,23 @@ pub fn to_starknet_transaction(
     }
 
     let mut calldata = Vec::with_capacity(capacity);
+
+    // assert that the selector < FieldElement::MAX - retries
+    assert!(*ETH_SEND_TRANSACTION < FieldElement::MAX - retries.into());
```

Comment:
this assertion is done at compile time?

Reply:
should be runtime :) since retries can only be known at runtime